internal/bminventory/inventory.go

```diff
@@ -1809,17 +1816,13 @@ func (b *bareMetalInventory) updateNonDhcpNetworkParams(updates map[string]inter
 				return common.NewApiError(http.StatusBadRequest, errors.Wrap(err, "Calculate machine network CIDR"))
 			}
 			if primaryMachineNetworkCidr != "" {
-				if network.IsMachineCidrAvailable(cluster) {
```

Comment:
It checks the same thing.
But the change checks it on the ClusterUpdateParams while before we checked it on cluster.
Since later we compare the ClusterUpdateParams and the cluster before updating, and then copy the parameters from ClusterUpdateParams to cluster, then here we just modify the ClusterUpdateParams and not the cluster.
Otherwise, if we modify here the cluster, then later when we check the difference (if to update), it will indicate that there was no change.